ivadomed/utils.py

```diff
@@ -236,14 +241,42 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
         nib_im = nib.load(fname)
         data_im = nib_im.get_fdata()
         data_lst.append(data_im)
-        data_im_l, _ = label(data_im, structure=bin_struct)
+        data_im_l = []
+        for i in range(data_im.shape[-1]):
+            data_im_tmp, _ = label(data_im[..., i], structure=bin_struct)
+            data_im_l.append(data_im_tmp)
         data_l_lst.append(data_im_l)
         del nib_im
+    data_l_lst = np.array(data_l_lst)
+    # channel went first due to the 'append' function
+    # Befor transpose we are supposed to have MC_simulation X Channel X Height X Width X Depth
+    # After transpose we expect to have MC_simulation X Height X Width X Depth X Channel
+    # this is wanted because the network output has a shape of  Channel X Height X Width X Depth which is saved as
+    # nifti with shape Height X Width X Depth X Channel. We are using the second shape to save image afterward.
+
+    if len(data_l_lst.shape) == 4:
+        data_l_lst = np.transpose(data_l_lst, (0, 2, 3, 4, 1))
+    elif len(data_l_lst.shape) == 3:
+        data_l_lst = np.transpose(data_l_lst, (0, 2, 3, 1))
+    else:
+        print("object does not appear to be 2D or 3D, please check the network output")
+        exit(1)
```

Comment:
use logging in API code-- is that already a written convention @Drulex ?
also, i think we should avoid these hard exit in APIs (instead, raise, and the parent function takes care of the rest)

Reply:
I don't see it in this project's wiki, but you can refer to sct's guidelines: https://github.com/neuropoly/spinalcordtoolbox/wiki more specifically the Logging and Exceptions sections.

Reply:
can someone pls update the wiki (maybe redirect to sct wiki for now until someone can dedicate the time to make proper wiki-- thx)

Reply:
Perhaps we should have an section on API code guidelines. Some generic things come to mind:
- No printing, use logging instead
- Don't use `exit`, raise and let parent handle
- Avoid system calls
- Deal with low-level objects instead of files
- Include unit tests
- Make interfaces generic and re-usable

Reply:
awesome! let's start with this-- thx!

Reply:
started in 65037df

Reply:
> awesome! let's start with this-- thx!

i meant the wiki-- can someone do it?

Reply:
I can take care of it

Reply:
WIP -> https://github.com/ivadomed/ivadomed/wiki/Coding-Best-Practices

Reply:
fantastic! thank you so much @Drulex -- we could use that template for other projects as well

Reply:
or maybe we should centralize (instead of duplicating across projects) such guidelines... 🤔  thoughts?

Reply:
There are could be some instruction specific to ivadomed (where to add code, how to name import (e.g., import ivadomed.utils as imed_utils), So I think the template approach makes more sens.